WordPress/gutenberg
# Add template revision button

## What?
Adds a button to templates and template parts to get to the revision compare screen

## Why?

Closes #44503

## How?

This adds a button to the site editor for Templates and Template Parts that links to the revision viewer for that template.

It adds 2 new fields to the REST response for `/wp/v2/template/{theme}//{part}` and `/wp/v2/template-part/{theme}//{part}` routes: 'revision_count' and `latest_revision_id`. The Posts REST controller uses the `_links` section to expose these and links to routes for the Revisions REST controller. I chose to add these to the body because the Revisions REST routes assume numeric IDs with `/revisions` at the end, and the existing template routes receive everything in their namespace. Adding `/revisions` routes for templates and template-parts would require complex RegExes that could potentially lead to a DoS attack. However if someone can get the routes on the Revisions REST controller to work with Template IDs, I would be happy to add them.

## Testing Instructions




1. Open the Site Editor
2. Edit a Template or Template Part
3. Save to create a custom template
4. Make another change and save again, to create a second custom template
5. Confirm that the revision button appears in the Template sidebar and links to the revision viewer

## Screenshots or screencast 

https://user-images.githubusercontent.com/107534/217497598-9b112429-2a03-4fc2-b2e2-c1a8537acf21.mp4
